src/main/java/com/aws/greengrass/componentmanager/builtins/ArtifactDownloader.java

```diff
@@ -250,5 +251,35 @@ protected String getErrorString(String reason) {
         return String.format(ARTIFACT_DOWNLOAD_EXCEPTION_FMT, artifact.getArtifactUri(),
                 identifier.getName(), identifier.getVersion().toString()) + reason;
     }
+
+    /**
+     * Check if an instance of implemented class supports checking component store size depending on
+     * if the artifact is located in greengrass artifact store or third party.
+     *
+     * @return evaluation result
+     */
+    public boolean checkComponentStoreSize() {
```

Comment:
"**can**CheckComponentStoreSize"?

Reply:
I meant for it to convey if if there's a point in checking component store size, we of course can check the component store size always but there's no use doing that if artifacts are located in a third party store like docker images